diff --git a/node_modules/@graphcommerce/magento-cart-shipping-address/components/CustomerAddressForm/CustomerAddressForm.tsx b/node_modules/@graphcommerce/magento-cart-shipping-address/components/CustomerAddressForm/CustomerAddressForm.tsx
index 6f4add4..f5be44e 100644
--- a/node_modules/@graphcommerce/magento-cart-shipping-address/components/CustomerAddressForm/CustomerAddressForm.tsx
+++ b/node_modules/@graphcommerce/magento-cart-shipping-address/components/CustomerAddressForm/CustomerAddressForm.tsx
@@ -44,7 +44,6 @@ export function CustomerAddressForm(props: CustomerAddressListProps) {
   const isVirtual = cart?.is_virtual ?? false
 
   const customerAddresses = filterNonNullableKeys(customer.data?.customer?.addresses, ['id'])
-
   const cartShipping = cart?.shipping_addresses?.[0]
   const cartShippingId =
     findCustomerAddressFromCartAddress(customerAddresses, cartShipping)?.id || null
diff --git a/node_modules/@graphcommerce/magento-cart-shipping-address/components/ShippingAddressForm/ShippingAddressForm.tsx b/node_modules/@graphcommerce/magento-cart-shipping-address/components/ShippingAddressForm/ShippingAddressForm.tsx
index a128fe9..b30f433 100644
--- a/node_modules/@graphcommerce/magento-cart-shipping-address/components/ShippingAddressForm/ShippingAddressForm.tsx
+++ b/node_modules/@graphcommerce/magento-cart-shipping-address/components/ShippingAddressForm/ShippingAddressForm.tsx
@@ -1,6 +1,7 @@
 import type { UseFormComposeOptions } from '@graphcommerce/ecommerce-ui'
 import {
   CheckboxElement,
+  ComposedForm,
   FormAutoSubmit,
   FormPersist,
   TelephoneElement,
@@ -24,14 +25,16 @@ import {
 import { CountryRegionsDocument, StoreConfigDocument } from '@graphcommerce/magento-store'
 import { Form, FormRow } from '@graphcommerce/next-ui'
 import { Trans } from '@lingui/macro'
-import type { SxProps, Theme } from '@mui/material'
-import React from 'react'
+import { Box, Typography, type SxProps, type Theme } from '@mui/material'
+import React, { useEffect } from 'react'
 import { isCartAddressACustomerAddress } from '../../utils/findCustomerAddressFromCartAddress'
 import { isSameAddress } from '../../utils/isSameAddress'
 import { GetAddressesDocument } from './GetAddresses.gql'
 import { SetBillingAddressDocument } from './SetBillingAddress.gql'
 import { SetShippingAddressDocument } from './SetShippingAddress.gql'
 import { SetShippingBillingAddressDocument } from './SetShippingBillingAddress.gql'
+import { EmailForm } from '@graphcommerce/magento-cart-email'
+import { ShippingMethodFormDocument } from '@graphcommerce/magento-cart-shipping-method'
 
 export type ShippingAddressFormProps = Pick<UseFormComposeOptions, 'step'> & {
   /**
@@ -39,10 +42,12 @@ export type ShippingAddressFormProps = Pick<UseFormComposeOptions, 'step'> & {
    */
   ignoreCache?: boolean
   sx?: SxProps<Theme>
+  isPickup?: boolean | undefined
 }
 
 export const ShippingAddressForm = React.memo<ShippingAddressFormProps>((props) => {
-  const { step, sx } = props
+  const { step, sx, isPickup = false } = props
+
   const { data: cartQuery } = useCartQuery(GetAddressesDocument)
   const { data: config } = useQuery(StoreConfigDocument)
   const countryQuery = useQuery(CountryRegionsDocument, { fetchPolicy: 'cache-and-network' })
@@ -86,24 +91,24 @@ export const ShippingAddressForm = React.memo<ShippingAddressFormProps>((props)
     defaultValues: isCartAddressACustomerAddress(customerQuery?.customer?.addresses, currentAddress)
       ? { saveInAddressBook: true, isCompany: false }
       : {
-          // todo(paales): change to something more sustainable
-          firstname: currentAddress?.firstname ?? customerQuery?.customer?.firstname ?? '',
-          lastname: currentAddress?.lastname ?? customerQuery?.customer?.lastname ?? '',
-          telephone:
-            currentAddress?.telephone !== '000 - 000 0000' ? currentAddress?.telephone : '',
-          city: currentAddress?.city ?? '',
-          company: currentAddress?.company ?? '',
-          vatId: currentAddress?.vat_id ?? '',
-          postcode: currentAddress?.postcode ?? '',
-          street: currentAddress?.street?.[0] ?? '',
-          houseNumber: currentAddress?.street?.[1] ?? '',
-          addition: currentAddress?.street?.[2] ?? '',
-          regionId: currentAddress?.region?.region_id ?? null,
-          countryCode: currentAddress?.country.code ?? shopCountry, // todo: replace by the default shipping country of the store + geoip,
-          saveInAddressBook: true,
-          isCompany: Boolean(currentAddress?.company || currentAddress?.vat_id),
-          customerNote: '',
-        },
+        // todo(paales): change to something more sustainable
+        firstname: currentAddress?.firstname ?? customerQuery?.customer?.firstname ?? '',
+        lastname: currentAddress?.lastname ?? customerQuery?.customer?.lastname ?? '',
+        telephone:
+          currentAddress?.telephone !== '000 - 000 0000' ? currentAddress?.telephone : '',
+        city: currentAddress?.city ?? '',
+        company: currentAddress?.company ?? '',
+        vatId: currentAddress?.vat_id ?? '',
+        postcode: currentAddress?.postcode ?? '',
+        street: currentAddress?.street?.[0] ?? '',
+        houseNumber: currentAddress?.street?.[1] ?? '',
+        addition: currentAddress?.street?.[2] ?? '',
+        regionId: currentAddress?.region?.region_id ?? null,
+        countryCode: currentAddress?.country.code ?? shopCountry, // todo: replace by the default shipping country of the store + geoip,
+        saveInAddressBook: true,
+        isCompany: Boolean(currentAddress?.company || currentAddress?.vat_id),
+        customerNote: '',
+      },
     skipUnchanged: true,
     onBeforeSubmit: (variables) => {
       const regionId = countries
@@ -114,7 +119,6 @@ export const ShippingAddressForm = React.memo<ShippingAddressFormProps>((props)
         variables.company = ''
         variables.vatId = ''
       }
-
       return {
         ...variables,
         telephone: variables.telephone || '000 - 000 0000',
@@ -124,32 +128,148 @@ export const ShippingAddressForm = React.memo<ShippingAddressFormProps>((props)
       }
     },
   })
-  const { handleSubmit, required, error } = form
-  const submit = handleSubmit(() => {})
+  const { handleSubmit, required, error, data } = form
+  const submit = handleSubmit(() => { })
 
   useFormCompose({ form, step, submit, key: 'ShippingAddressForm' })
 
+
+  const method_form = useFormGqlMutationCart(ShippingMethodFormDocument, {
+    onBeforeSubmit: (variables: any) => {
+      const [carrier, method] = (variables?.carrierMethod ?? '').split('-')
+
+      if (carrier === "instore") {
+        return { ...variables, carrier, method: "pickup" }
+      }
+
+      if (carrier === "cityrate") {
+        return { ...variables, carrier, method: "cityrate" }
+      }
+
+      if (carrier === "flatrate") {
+        return { ...variables, carrier, method: "flatrate" }
+      }
+
+      return { ...variables, carrier, method }
+    }
+  })
+
+  const { handleSubmit: handleSubmitMethod, setValue } = method_form
+  const methodSubmit = handleSubmitMethod(() => { })
+
+  const setShippingMethod = (code, method) => {
+    setValue(code, method, { shouldValidate: true })
+    methodSubmit()
+  }
+
+  useEffect(() => {
+    if (data?.setShippingAddressesOnCart?.cart && form?.formState?.isSubmitSuccessful && !form?.formState?.isSubmitting && form?.formState?.isSubmitted) {
+      const shippingAddress = data?.setShippingAddressesOnCart?.cart?.shipping_addresses?.[0]
+      const availableMethods = (shippingAddress?.available_shipping_methods ?? [])
+
+      let selectedMethod
+
+      if (availableMethods?.length) {
+        if (isPickup) {
+          selectedMethod = availableMethods?.find(
+            (m: any) => (m?.carrier_code === 'instore' && m?.method_code === 'pickup') || (m?.carrier_code === "cityrate" && m?.method_code === "cityrate") || availableMethods?.[0],
+          )
+        } else {
+          const deliveryMethods = availableMethods?.filter(
+            (m: any) => !(m?.carrier_code === 'instore' && m?.method_code === 'pickup'),
+          )
+
+          if (deliveryMethods?.length) {
+            selectedMethod = deliveryMethods?.find(
+              (m: any) => m?.carrier_code === "cityrate" && m?.method_code === "cityrate",
+            ) || deliveryMethods?.[0]
+          }
+        }
+
+        if (selectedMethod) {
+          setShippingMethod("carrierMethod", selectedMethod?.carrier_code)
+        }
+      }
+    }
+  }, [isPickup, form?.formState?.isSubmitSuccessful, form?.formState?.isSubmitting, form?.formState?.isSubmitted])
+
   return (
     <Form onSubmit={submit} noValidate sx={sx}>
       <FormAutoSubmit
         submit={submit}
         control={form.control}
-        name={['postcode', 'countryCode', 'regionId']}
+        name={['postcode', 'countryCode', 'regionId', 'city']}
       />
 
       <CompanyFields form={form} />
       <NameFields form={form} />
-      <AddressFields form={form} />
-
       <FormRow>
-        <TelephoneElement
-          control={form.control}
-          name='telephone'
-          variant='outlined'
-          required={required.telephone}
-          showValid
+        <EmailForm
+          step={1}
+          sx={{
+            '& .MuiInputLabel-formControl': {
+              color: (theme) => theme.palette.custom.main,
+              //    marginBottom:'clamp(10px, (9.14px + 0.27vw), 16px)',
+              fontSize: { xs: '15px', md: '16px' },
+              lineHeight: '158%',
+              fontWeight: 400,
+
+              '&.Mui-focused': {
+                color: (theme) => theme.palette.custom.main,
+              },
+              '& .MuiFormLabel-asterisk': {
+                display: 'none',
+              },
+            },
+
+            '& .MuiOutlinedInput-root': {
+              border: (theme) => `1px solid ${theme.palette.custom.border}`,
+              borderRadius: '4px',
+
+              '&:hover': {
+                border: (theme) => `1px solid ${theme.palette.custom.main}`,
+              },
+              '&.Mui-focused': {
+                border: (theme) => `1px solid ${theme.palette.custom.main}`,
+              },
+
+              '& .MuiOutlinedInput-notchedOutline': {
+                border: 'none',
+              },
+            },
+          }}
         />
+
+        <FormRow sx={{
+          paddingTop: 0,
+          paddingBottom: 0,
+        }}>
+          <TelephoneElement
+            control={form.control}
+            name='telephone'
+            variant='outlined'
+            required={required.telephone}
+            showValid
+          />
+        </FormRow>
       </FormRow>
+
+      <Box>
+        <Typography sx={{
+          color: (theme) => theme.palette.custom.dark,
+          fontSize: { xs: '16px', md: '20px' },
+          marginTop: '15px',
+          marginBottom: { xs: '10px', md: '15px' },
+          textTransform: 'capitalize',
+          fontWeight: 400,
+          fontVariationSettings: '"wght" 400',
+        }}>Your Address</Typography>
+
+        <AddressFields form={form} />
+      </Box>
+
+
+
       {customerQuery?.customer && (
         <CheckboxElement
           control={form.control}
