diff --git a/node_modules/@graphcommerce/magento-store/Money.tsx b/node_modules/@graphcommerce/magento-store/Money.tsx
index 869323a..2cc64df 100644
--- a/node_modules/@graphcommerce/magento-store/Money.tsx
+++ b/node_modules/@graphcommerce/magento-store/Money.tsx
@@ -3,16 +3,18 @@ import type { CurrencyFormatProps } from '@graphcommerce/next-ui'
 import { CurrencyFormat } from '@graphcommerce/next-ui'
 import type { MoneyFragment } from './Money.gql'
 import { StoreConfigDocument } from './StoreConfig.gql'
+import type { SxProps, Theme } from '@mui/material'
 
 type OverridableProps = {
   round?: boolean
   formatOptions?: Omit<CurrencyFormatProps, 'currency'>
+  sx?: SxProps<Theme>
 }
 
 export type MoneyProps = MoneyFragment & OverridableProps
 
 export function Money(props: MoneyProps) {
-  const { currency, value, round = false, formatOptions } = props
+  const { currency, value, round = false, formatOptions ,sx} = props
   const baseCurrencyCode = useQuery(StoreConfigDocument).data?.storeConfig?.base_currency_code
   const digits = (value ?? 0) % 1 !== 0
   const maximumFractionDigits = round && !digits ? 0 : 2
@@ -26,6 +28,7 @@ export function Money(props: MoneyProps) {
       maximumFractionDigits={maximumFractionDigits}
       {...formatOptions}
       value={value}
+      sx={sx}
     />
   )
 }
diff --git a/node_modules/@graphcommerce/magento-store/StoreConfigFragment.graphql b/node_modules/@graphcommerce/magento-store/StoreConfigFragment.graphql
index c7227bc..bb4edb8 100644
--- a/node_modules/@graphcommerce/magento-store/StoreConfigFragment.graphql
+++ b/node_modules/@graphcommerce/magento-store/StoreConfigFragment.graphql
@@ -35,4 +35,6 @@ fragment StoreConfigFragment on StoreConfig {
   order_cancellation_reasons {
     description
   }
+
+  customer_access_token_lifetime
 }
