diff --git a/node_modules/@graphcommerce/ecommerce-ui/components/FormComponents/ActionCardListForm.tsx b/node_modules/@graphcommerce/ecommerce-ui/components/FormComponents/ActionCardListForm.tsx
index 1834408..d837058 100644
--- a/node_modules/@graphcommerce/ecommerce-ui/components/FormComponents/ActionCardListForm.tsx
+++ b/node_modules/@graphcommerce/ecommerce-ui/components/FormComponents/ActionCardListForm.tsx
@@ -1,12 +1,18 @@
+import { CheckboxElement } from '@graphcommerce/ecommerce-ui/components/FormComponents/CheckboxElement'
 import type { ActionCardListProps, ActionCardProps } from '@graphcommerce/next-ui'
 import { ActionCardList } from '@graphcommerce/next-ui'
 import type { ControllerProps, FieldValues } from '@graphcommerce/react-hook-form'
-import { useController } from '@graphcommerce/react-hook-form'
+import { useController, useForm } from '@graphcommerce/react-hook-form'
+import { Box, Checkbox, FormControl, FormControlLabel, FormGroup, InputLabel, Typography, useForkRef } from '@mui/material'
 import type { MouseEventHandler } from 'react'
-import React, { useCallback } from 'react'
+import React, { useCallback, useRef } from 'react'
 
 export type ActionCardItemBase = Pick<ActionCardProps, 'value'>
 
+export type  actionCadListFormProps = {
+  isFilter?:boolean
+}
+
 export type ActionCardRequireOptionSelection = { requireOptionSelection?: boolean }
 
 export type ActionCardItemRenderProps<T> = ActionCardProps & {
@@ -18,9 +24,9 @@ export type ActionCardListFormProps<A, F extends FieldValues = FieldValues> = Om
   'value' | 'error' | 'onChange' | 'children'
 > &
   Omit<ControllerProps<F>, 'render'> & {
-    items: A[]
+    items: A[] | any
     render: React.FC<ActionCardItemRenderProps<A>>
-  } & ActionCardRequireOptionSelection
+  } & ActionCardRequireOptionSelection & actionCadListFormProps
 
 /** @public */
 export function ActionCardListForm<
@@ -40,6 +46,7 @@ export function ActionCardListForm<
     disabled,
     shouldUnregister,
     requireOptionSelection,
+    isFilter,
     ...other
   } = props
   const RenderItem = render as React.FC<ActionCardItemRenderProps<ActionCardItemBase>>
@@ -53,9 +60,14 @@ export function ActionCardListForm<
     [multiple],
   )
 
+  // const inputRef = useRef(null);
+  // const form  = useForm();
+
+
   const {
-    field: { onChange, value, ref },
+    field: { onChange, value, ref,...field },
     fieldState,
+    fieldState:{invalid, error},
     formState,
   } = useController({
     ...props,
@@ -75,6 +87,9 @@ export function ActionCardListForm<
     [onChange, requireOptionSelection],
   )
 
+  // console.log('ActionCardListForm', items)
+
+
   return (
     <ActionCardList
       {...other}
@@ -86,6 +101,30 @@ export function ActionCardListForm<
       error={formState.isSubmitted && !!fieldState.error}
       errorMessage={fieldState.error?.message}
     >
+      {/* {items.map((item) => (
+        <Box>
+          <Typography component={"label"}>{item.label}</Typography>
+          <Checkbox
+            key={`${item.value}`}
+            title={item.label}
+            value={item.value}
+            checked={onSelect(item.value, value)}
+          />
+        </Box>
+      ))} 
+
+{ /*items.map((item) => (
+         <CheckboxElement
+      label={item.label}
+      value={item.label}
+      checked={onSelect(item.value, value)}
+      name={"sort"}
+      control={control}
+      
+    /> 
+      )) */}
+   
+      
       {items.map((item) => (
         <RenderItem
           {...item}
@@ -93,8 +132,9 @@ export function ActionCardListForm<
           value={item.value}
           selected={onSelect(item.value, value)}
           onReset={handleReset}
+          control={control}
         />
-      ))}
+      )) }
     </ActionCardList>
   )
 }
diff --git a/node_modules/@graphcommerce/ecommerce-ui/components/FormComponents/CheckboxElement.tsx b/node_modules/@graphcommerce/ecommerce-ui/components/FormComponents/CheckboxElement.tsx
index d6afcc5..d396279 100644
--- a/node_modules/@graphcommerce/ecommerce-ui/components/FormComponents/CheckboxElement.tsx
+++ b/node_modules/@graphcommerce/ecommerce-ui/components/FormComponents/CheckboxElement.tsx
@@ -21,7 +21,10 @@ export type CheckboxElementProps<T extends FieldValues> = Omit<CheckboxProps, 'n
   label?: FormControlLabelProps['label']
   helperText?: string
   sx?: SxProps<Theme>
+  onReset?: (event: React.MouseEvent<HTMLElement>) => void
   formControl?: Omit<FormControlProps<'div'>, 'required' | 'error'>
+  checkboxColor?: string | ((theme: Theme) => string) // For unchecked state
+  checkboxCheckedColor?: string | ((theme: Theme) => string) // For checked state
 } & Omit<ControllerProps<T>, 'render'>
 
 /** @public */
@@ -40,6 +43,8 @@ export function CheckboxElement<TFieldValues extends FieldValues>(
     defaultValue,
     disabled,
     shouldUnregister,
+    checkboxColor,
+    checkboxCheckedColor,
     ...rest
   } = props
 
@@ -72,7 +77,14 @@ export function CheckboxElement<TFieldValues extends FieldValues>(
               {...field}
               inputRef={useForkRef(ref, rest.inputRef)}
               color={rest.color || 'primary'}
-              sx={{
+              sx={{'&.MuiCheckbox-root': {
+                  color: "#F1A8B6 !important", 
+                  // padding:'0 9px 0 0 !important',
+
+                  '&:hover': {
+                    backgroundColor: 'transparent !important',
+                  }
+                },
                 ...(Array.isArray(sx) ? sx : [sx]),
                 color: invalid ? 'error.main' : undefined,
               }}
