diff --git a/node_modules/@graphcommerce/magento-cart/components/CartFab/CartFab.tsx b/node_modules/@graphcommerce/magento-cart/components/CartFab/CartFab.tsx
index 2cb4e6f..567bda0 100644
--- a/node_modules/@graphcommerce/magento-cart/components/CartFab/CartFab.tsx
+++ b/node_modules/@graphcommerce/magento-cart/components/CartFab/CartFab.tsx
@@ -64,19 +64,32 @@ function CartFabContent(props: CartFabContentProps) {
         aria-label={i18n._(/* i18n */ 'Cart')}
         color='inherit'
         size='responsive'
-        style={{ backgroundColor }}
+        // style={{ backgroundColor }}
         sx={(theme) => ({
           [theme.breakpoints.down('md')]: {
             backgroundColor: `${theme.palette.background.paper} !important`,
           },
+          height:"40px !important"
         })}
         {...fabProps}
       >
         <DesktopHeaderBadge
-          color='primary'
-          variant='dot'
-          overlap='circular'
-          badgeContent={total_quantity}
+          sx={{
+    '& .MuiBadge-badge': { 
+      backgroundColor: '#FF0000',
+      color:'#fff',
+      display:'flex',
+      alignItems:'center',
+     justifyContent:'center',
+     textAlign:'center',
+     fontSize:'9px',
+     minWidth:'12px',
+     height:'14px',
+     right:'16px',
+     top:'4px'
+    },
+  }} 
+          badgeContent={total_quantity} overlap='circular'
           {...BadgeProps}
         >
           {cartIcon}
@@ -91,7 +104,7 @@ function CartFabContent(props: CartFabContentProps) {
           position: 'absolute',
           height: '100%',
           width: '100%',
-          boxShadow: 6,
+          // boxShadow: 6,
           top: 0,
           [theme.breakpoints.down('md')]: {
             opacity: '1 !important',
diff --git a/node_modules/@graphcommerce/magento-cart/components/CartStartCheckout/CartStartCheckout.tsx b/node_modules/@graphcommerce/magento-cart/components/CartStartCheckout/CartStartCheckout.tsx
index 0b04f2c..5a63da3 100644
--- a/node_modules/@graphcommerce/magento-cart/components/CartStartCheckout/CartStartCheckout.tsx
+++ b/node_modules/@graphcommerce/magento-cart/components/CartStartCheckout/CartStartCheckout.tsx
@@ -13,6 +13,8 @@ export type CartStartCheckoutProps = {
   buttonProps?: ButtonProps<'button'>
   disabled?: boolean
   cart?: CartStartCheckoutFragment | null | undefined
+  title?:string
+  isCheckout?:boolean
   onStart?: (
     e: React.MouseEvent<HTMLButtonElement>,
     cart: CartStartCheckoutFragment | null | undefined,
@@ -37,6 +39,8 @@ export function CartStartCheckout(props: CartStartCheckoutProps) {
     disabled,
     sx = [],
     cart,
+    title='Proceed to Checkout',
+    isCheckout=false
   } = props
 
   const shouldLoginToContinue = useCheckoutShouldLoginToContinue()
@@ -63,13 +67,13 @@ export function CartStartCheckout(props: CartStartCheckoutProps) {
       )}
 
       <Button
-        href='/checkout'
+        href={isCheckout?'/checkout/payment':'/checkout'}
         id='cart-start-checkout'
         variant='pill'
         color='secondary'
         size='large'
-        className={classes.checkoutButton}
-        endIcon={<IconSvg src={iconChevronRight} />}
+        // className={classes.checkoutButton}
+        // endIcon={<IconSvg src={iconChevronRight} />}
         onClick={(e) => {
           onClick?.(e)
           onStart?.(e, cart)
@@ -86,13 +90,13 @@ export function CartStartCheckout(props: CartStartCheckoutProps) {
             '& ~ span.MuiButton-endIcon': { marginLeft: '6px' },
           })}
         >
-          <Trans>Start Checkout</Trans>
+          {title}
         </Box>{' '}
-        {hasTotals && (
+        {/*hasTotals && (
           <span className={classes.checkoutMoney}>
             <Money {...cart?.prices?.grand_total} />
           </span>
-        )}
+        )*/}
       </Button>
 
       {children}
diff --git a/node_modules/@graphcommerce/magento-cart/components/CartSummary/CartSummary.tsx b/node_modules/@graphcommerce/magento-cart/components/CartSummary/CartSummary.tsx
index 0d3023f..53aaeb0 100644
--- a/node_modules/@graphcommerce/magento-cart/components/CartSummary/CartSummary.tsx
+++ b/node_modules/@graphcommerce/magento-cart/components/CartSummary/CartSummary.tsx
@@ -102,7 +102,7 @@ export function CartSummary(props: CartSummaryProps) {
             />
             <Typography variant='body1'>
               {shipping_addresses?.[0]?.selected_shipping_method?.carrier_title}
-              {shipping_addresses?.[0]?.selected_shipping_method?.method_title}
+              {/* {shipping_addresses?.[0]?.selected_shipping_method?.method_title} */}
             </Typography>
           </Box>
         )}
diff --git a/node_modules/@graphcommerce/magento-cart/components/CartTotals/CartTotals.tsx b/node_modules/@graphcommerce/magento-cart/components/CartTotals/CartTotals.tsx
index 11b4847..251f6cb 100644
--- a/node_modules/@graphcommerce/magento-cart/components/CartTotals/CartTotals.tsx
+++ b/node_modules/@graphcommerce/magento-cart/components/CartTotals/CartTotals.tsx
@@ -2,7 +2,7 @@ import { Money } from '@graphcommerce/magento-store'
 import { breakpointVal, extendableComponent } from '@graphcommerce/next-ui'
 import { Trans } from '@lingui/react'
 import type { SxProps, Theme } from '@mui/material'
-import { Box, Divider, lighten } from '@mui/material'
+import { Box, Divider, lighten, Typography } from '@mui/material'
 import { useCartQuery, useDisplayInclTax } from '../../hooks'
 import { GetCartTotalsDocument } from './GetCartTotals.gql'
 
@@ -85,10 +85,22 @@ export function CartTotals(props: CartTotalsProps) {
           key='subtotal'
           sx={{ display: 'flex', justifyContent: 'space-between' }}
         >
-          <Box>
-            <Trans id='Products' />
+          <Box sx={{
+            color: (theme: any) => theme.palette.custom.smallHeading,
+            fontSize: { xs: '15px', md: '16px' },
+            fontWeight: 400,
+            lineHeight: 'normal',
+          }}>
+            <Trans id='Items Total' />
           </Box>
-          <Box className={classes.money} sx={{ whiteSpace: 'nowrap' }}>
+          <Box className={classes.money} sx={{ whiteSpace: 'nowrap',
+            '& span span':{
+               color: (theme: any) => theme.palette.custom.smallHeading,
+               fontWeight: 600,
+               fontSize: { xs: '15px', md: '16px' },
+               textTransform: 'capitalize',
+            }
+           }}>
             <Money {...(inclTax ? prices.subtotal_including_tax : prices.subtotal_excluding_tax)} />
           </Box>
         </Box>
@@ -108,10 +120,32 @@ export function CartTotals(props: CartTotalsProps) {
             sx={{
               display: 'flex',
               justifyContent: 'space-between',
+              alignItems:'center'
             }}
           >
-            <Box>{discount?.label}</Box>
-            <Box className={classes.money} sx={{ whiteSpace: 'nowrap' }}>
+            <Box>
+              <Typography sx={{
+            color: (theme: any) => theme.palette.custom.smallHeading,
+            fontSize: { xs: '15px', md: '16px' },
+            fontWeight: 400,
+            lineHeight: 'normal',
+          }}>
+                Discount
+              </Typography>
+              <Typography sx={{
+                fontSize:{xs:'15px', md:'15px'},
+                color:'#9d9d9d',
+                marginTop:'2px'
+              }}>
+              {discount?.label && `(${discount.label})`}
+              </Typography>
+            </Box>
+            <Box  className={`${classes.money} discount`} sx={{ whiteSpace: 'nowrap','& span span':{
+               color: (theme: any) => theme.palette.custom.smallHeading,
+               fontWeight: 600,
+               fontSize: { xs: '15px', md: '16px' },
+               textTransform: 'capitalize',
+            } }}>
               {discount?.amount && <Money {...discount.amount} value={value} />}
             </Box>
           </Box>
@@ -154,6 +188,8 @@ export function CartTotals(props: CartTotalsProps) {
           </Box>
         ))}
 
+       
+
       {additionalSubtotals}
 
       <Box key='divider'>
@@ -170,10 +206,22 @@ export function CartTotals(props: CartTotalsProps) {
             color: theme.palette.primary.main,
           })}
         >
-          <Box>
+          <Box sx={{
+              color: (theme: any) => theme.palette.custom.smallHeading,
+              fontSize: { xs: '16px', md: '20px' },
+              fontWeight: 700,
+              lineHeight: 'normal',
+              textTransform: 'capitalize',
+          }}>
             <Trans id='Grand total' />
           </Box>
-          <Box className={classes.money} sx={{ whiteSpace: 'nowrap' }}>
+          <Box className={classes.money} sx={{ whiteSpace: 'nowrap' , 
+          '& span span': {
+           color: (theme: any) => theme.palette.custom.smallHeading,
+           fontWeight: 700,
+           fontSize: { xs: '16px', md: '20px' },
+           textTransform: 'capitalize',
+                      },}}>
             <Money {...prices.grand_total} />
           </Box>
         </Box>
@@ -201,6 +249,7 @@ export function CartTotals(props: CartTotalsProps) {
         ))}
 
       {additionalTotals}
+      <Typography sx={{color:'#8f8f8f', fontSize:'16px'}}>Proceed to checkout and add location to see shipping charges</Typography>
     </Box>
   )
 }
diff --git a/node_modules/@graphcommerce/magento-cart/hooks/useFormGqlMutationCart.ts b/node_modules/@graphcommerce/magento-cart/hooks/useFormGqlMutationCart.ts
index 4eb01c1..8fecb12 100644
--- a/node_modules/@graphcommerce/magento-cart/hooks/useFormGqlMutationCart.ts
+++ b/node_modules/@graphcommerce/magento-cart/hooks/useFormGqlMutationCart.ts
@@ -35,6 +35,7 @@ export function useFormGqlMutationCart<
     {
       ...options,
       onBeforeSubmit: async (variables) => {
+        // console.log(variables,options,"==>variables")
         if (shouldLoginToContinue && shouldBlockOperation) {
           return false
         }
